include/odp/api/spec/ipsec.h

```diff
@@ -772,6 +773,75 @@ typedef struct odp_ipsec_sa_param_t {
 
 } odp_ipsec_sa_param_t;
 
+/**
+ * IPSEC SA information
+ */
+typedef struct odp_ipsec_sa_info_t {
+	/** Copy of IPSEC Security Association (SA) parameters */
+	odp_ipsec_sa_param_t param;
+
+	/** IPSEC SA direction dependent parameters */
+	union {
+		/** Inbound specific parameters */
+		struct {
+			/** Additional SA lookup parameters. */
+			struct {
+				/** IP destination address (NETWORK ENDIAN) to
+				 *  be matched in addition to SPI value. */
+				uint8_t dst_addr[ODP_IPV6_ADDR_SIZE];
+			} lookup_param;
+
+			/** Antireplay window size
+			 *
+			 * Antireplay window size configured for the SA.
+			 * This value can be different from what application
+			 * had requested.
+			 */
+			uint32_t antireplay_ws;
+
+			/** Antireplay window top
+			 *
+			 * Sequence number representing a recent top of the
+			 * antireplay window(need not be the latest).
+			 * When no packets have been processed, this value will
+			 * be zero.
+			 */
+			uint64_t antireplay_window_top;
```

Comment:
This should also tell what is returned if the anti-replay service is disabled. There is a space missing before the opening parenthesis. Maybe this could be worded like this:

Sequence number representing a recent top of the anti-replay window. There may be a delay before the SA state is reflected in the value. The value will be zero if no packets have been processed or if the anti-replay service is not enabled.

Reply:
Ack.